src/intro-to-c/debugging/gdb-intro.md

```diff
@@ -157,9 +165,13 @@ $3 = (int *) 0x4
 (gdb)
 ```
 
-Okay, so it happened on my first time through the loop, and `current` *clearly* isn't pointing to `arr`. I'll continue and see if anything else pops up.
+> The `print` command can print specific values to the screen. You can use it to get an idea of what your program looked like when it stopped.
 
-```
+Okay, since `i` is 0, that means the segfault happened on my first time through the loop. I can also see that `current` *clearly* isn't pointing to `arr`'s address.
+
+In fact, the address in `current` is only 4 bytes away from `NULL`, so it looks like I dereferenced a `NULL` pointer then did the post-increment. I'll continue and see if anything else pops up.
```

Comment:
Does this explanation about `current` make sense? I'm wondering if it's a little convoluted.

If it is, then I could update the source code so that the dereferencing and post-incrementing happen on separate lines. That way, when I print `current`, we would clearly see that it was pointing to 0x0 when I dereferenced it.

Also, I noticed that the [Operators file]https://github.com/wikiknights/cs-materials/blob/main/src/intro-to-c/operators/operators-intro.md#arithmetic-operators) mentions the increment (`++`) and decrement (`--`) operators but doesn't explicitly mention pre- and post-incrementing. Is that something you think we should add?

Reply:
I see what you mean, that is something I am sure you can take on, so I won't say anything about the lines on `current`.

For the operators, that is a good point, and I would add in something about pre- and post-incrementing if you are able.